pkg/k8s/apis/cilium.io/v2/cnp_types.go

```diff
@@ -218,6 +218,19 @@ func (r *CiliumNetworkPolicy) Parse() (api.Rules, error) {
 	}
 
 	namespace := k8sUtils.ExtractNamespace(&r.ObjectMeta)
+	// Temporary fix for CCNPs. See #12834.
+	// TL;DR. CCNPs are converted into SlimCNPs and end up here so we need to
+	// convert them back to CCNPs to allow proper parsing.
+	if namespace == "" {
+		// return nil, fmt.Errorf("r.TypeMeta.Kind: %s", r.TypeMeta.Kind)
```

Comment:
leftover?

Reply:
:facepalm: Yes. Initial idea was to use `r.TypeMeta.Kind` to check CNP is actually a CCNP, but that's empty. I removed the comment.